packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts

```diff
@@ -1,33 +1,245 @@
 /*
- * Copyright (c) 2017, salesforce.com, inc.
- * All rights reserved.
- * Licensed under the BSD 3-Clause license.
- * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
- */
+* Copyright (c) 2017, salesforce.com, inc.
+* All rights reserved.
+* Licensed under the BSD 3-Clause license.
+* For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+*/
 import * as fs from 'fs';
 import * as path from 'path';
-import { SObjectCategory, SObjectDescribe } from '../describe';
+
+import { EOL } from 'os';
+import { mkdir, rm } from 'shelljs';
+
+import {
+  ChildRelationship,
+  Field,
+  SObject,
+  SObjectCategory,
+  SObjectDescribe
+} from '../describe';
 
 export class FauxClassGenerator {
-  public async generate(projectPath: string, type: SObjectCategory) {
+  private SFDX_DIR = '.sfdx';
+  private TOOLS_DIR = 'tools';
+  private SOBJECTS_DIR = 'sobjects';
+  private STANDARDOBJECTS_DIR = 'standardObjects';
+  private CUSTOMOBJECTS_DIR = 'customObjects';
+
+  public async generate(
```

Comment:
@GaryFNC  - What is your strategy to ensure _completeness_ of what you are doing? Did you go through each case at https://help.salesforce.com/articleView?id=overview_of_custom_object_relationships.htm&type=0 and handle each relationship?

Reply:
looking at that doc this morning - I have looked previously, but need to review it again

Reply:
I will make sure that the test plan covers all these situations and enter child bugs for anything not handled.   The primary cases are handled, but need to verify some of the more unusual cases